src/Engines/Native/nativeDataStream.ts

```diff
@@ -0,0 +1,115 @@
+export interface NativeData extends Uint32Array {
+}

```

Comment:
```suggestion
export type NativeData = Uint32Array;
```

Reply:
It's a bit weird that `NativeData` isn't used in this file. Maybe move this to nativeEngine and then you wouldn't have to export it.

Reply:
It is used in this file; there's a `writeNativeData` method. The intent by defining it here and providing that API is that this file can change what `NativeData` is under the hood (into a something as simple as a number or as complex as a JS type) without nativeEngine.ts having to change at all.